pandas/tests/frame/methods/test_explode.py

```diff
@@ -280,3 +280,24 @@ def test_multi_columns(input_subset, expected_dict, expected_index):
     result = df.explode(input_subset)
     expected = pd.DataFrame(expected_dict, expected_index)
     tm.assert_frame_equal(result, expected)
+
+
+def test_multi_columns2():
+    # GH 46084
+    df = pd.DataFrame(
+        {
+            "A": [[0, 1], [5], [], [2, 3]],
+            "B": [9, 8, 7, 6],
+            "C": [[1, 2], np.nan, [], [3, 4]],
+        }
+    )
+    result = df.explode(["A", "C"])
+    expected = pd.DataFrame(
+        {
+            "A": [0, 1, 5, np.nan, 2, 3],
+            "B": [9, 9, 8, 7, 6, 6],
+            "C": [1, 2, np.nan, np.nan, 3, 4],
+        },
+        index=[0, 0, 1, 2, 3, 3],
+    ).astype({"A": object, "C": object})
```

Comment:
Can you avoid the astype call?

Reply:
done. used np.array to set dtype of the cols